src/name/resolve.js

```diff
@@ -1,13 +1,16 @@
 'use strict'
 
+const ndjson = require('iterable-ndjson')
 const configure = require('../lib/configure')
+const toIterable = require('stream-to-it/source')
 
 module.exports = configure(({ ky }) => {
-  return async (path, options) => {
+  return async function * (path, options) {
     options = options || {}
 
     const searchParams = new URLSearchParams(options.searchParams)
     searchParams.set('arg', path)
+    searchParams.set('stream', true)
```

Comment:
the buffered version isn't an option anymore ?

Reply:
This yields increasingly more accurate values as they are discovered until the best value is selected from the quorum of 16. I decided to not give the user the option to `stream` here because even if it's set to false we'd still return an async iterable that yields only 1 result.

The quorum may never be reached so it's better to give a reault than fail entirely.

If people actually want the last (and best) value they can just use `it-last`:

```js
await last(ipfs.name.resolve('/ipns/QmHash'))
```

I will update the breaking change message to include this info.

Reply:
only if the resolve is done with the dht!.
it can be done by dnslink or other Routers like ipns over dns.

we have been discussing this stream option as the way to listen to secondary routers while already having a result to work with. Allowing instant results even if stalled and when a better result arrives do something, think service worker cache. 

forcing everyone to use iterator helpers feels weird, users don't always need streaming, users need a way to do streaming when they need to.

Reply:
![Screenshot 2019-12-12 at 16 01 48](https://user-images.githubusercontent.com/314190/70728189-db9a2180-1cf8-11ea-9cd2-898a05583775.png)
can't this be optional like go has in the cli ? "you need stream?" cool just pass stream: true and you get a stream.

Reply:
If it's not streaming by default users will just use the non-streaming APIs and be frustrated when things take a long time to resolve and they get no feedback. We need to build APIs that encourage streaming and discourage buffering. There's no downside if a result is available immeidately and it is streamed to you but if there is the possibility it needs to be sourced from multiple peers or is large or may take a while then we need to stream the result to give users feedback asap, if only to inform them of progress.